### PR TITLE
TK-1582: Added CI workflow for plugins

### PR DESCRIPTION
**Pull request by msurudwad**
Added CI workflow for kubectl-plugins

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,156 @@
+# CI WorkFlow for build, test and release of Plugin Packages
+name: CI for Plugin Packages
+on:
+  push:
+    branches:
```
Comment (shekhawatsanjay) [inline .github/workflows/ci.yml:5]: we can remove `main` from here 
just trigger on tag

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      - 'main'
+    tags:
+      - 'v*.*.*'
+    paths-ignore:
```
Comment (shekhawatsanjay) [inline .github/workflows/ci.yml:9]: remove path filters for `tag` release flow

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+      - 'plugins/**'
+      - 'hack/test-plugin-manifests.sh'
+      - 'hack/update-plugin-manifests.sh'
+      - 'hack/update-preflight-manifest.sh'
+      - 'hack/update-log-collector-manifest.sh'
+      - '.github/workflows/plugins.yml'
+  pull_request:
+    branches:
```
Comment (shekhawatsanjay) [inline .github/workflows/ci.yml:17]: only keep `main` for now here

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        uses: reviewdog/action-yamllint@v1
+        with:
+          fail_on_error: true
+          reporter: local
+          filter_mode: nofilter
+
+      - name: Run golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          version: v1.29
```
Comment (shekhawatsanjay) [inline .github/workflows/ci.yml:47]: change to 1.30

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+              - 'cmd/**'
+              - 'hack/**'
+              - '.krew/tvk-log-collector.yaml'
+              - 'tools/log_collector/**'
+              - '.goreleaser.yml'
+
+  preflight:
+    runs-on: ubuntu-latest
+    needs: pre-build
+    if: (needs.pre-build.outputs.preflight == 'true') || startsWith(github.ref, 'refs/tags/')
```
Comment (shekhawatsanjay) [inline .github/workflows/ci.yml:76]: add a comment as to why we need both of these conditions

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+
+      - name: Run golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          version: v1.29
+
+      - name: Run Shell lint
+        run: hack/run-shell-lint.sh
+
+      - name: Verify code patterns
```
Comment (shekhawatsanjay) [inline .github/workflows/ci.yml:52]: check if we can remove this and do these things via golangci-lint

File: .goreleaser.yml [hunk 7]
```diff
@@ -0,0 +1,51 @@
+before:
+  hooks:
+    - go mod download
```
Comment (shekhawatsanjay) [inline .goreleaser.yml:3]: use vendor/ instead of go mod download

File: .goreleaser.yml [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    - darwin
+    - windows
+  goarch:
+    - amd64
+
+archives:
+- id: log-collector-archive
+  name_template: |-
+    log-collector_{{ .Tag }}_{{ .Os }}_{{ .Arch -}}
+    {{- with .Arm -}}
```
Comment (shekhawatsanjay) [inline .goreleaser.yml:22]: we can remove `arm` part

File: .goreleaser.yml (hunk 7 above)
Comment (msurudwad) [reply .goreleaser.yml:3]: sure.. will remove this part of code.

File: .github/workflows/ci.yml (hunk 6 above)
Comment (msurudwad) [reply .github/workflows/ci.yml:52]: sure.. will do that

File: .github/workflows/ci.yml (hunk 1 above)
Comment (msurudwad) [reply .github/workflows/ci.yml:5]: yes.. done. Please review

File: .github/workflows/ci.yml (hunk 2 above)
Comment (msurudwad) [reply .github/workflows/ci.yml:9]: yes.. done. Please review

File: .github/workflows/ci.yml (hunk 4 above)
Comment (msurudwad) [reply .github/workflows/ci.yml:47]: yes.. done. Please review

File: .github/workflows/ci.yml (hunk 5 above)
Comment (msurudwad) [reply .github/workflows/ci.yml:76]: yes.. done. Please review

File: .goreleaser.yml (hunk 8 above)
Comment (msurudwad) [reply .goreleaser.yml:22]: yes.. done. Please review

File: .github/workflows/ci.yml (hunk 3 above)
Comment (msurudwad) [reply .github/workflows/ci.yml]: yes.. done. Please review

File: .github/workflows/plugin-packages.yml [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+        with:
+          go-version: 1.15
+
+      - name: Build Log-Collector Packages
+        uses: goreleaser/goreleaser-action@v2
+        with:
+          version: latest
+          args: release --snapshot --skip-publish --rm-dist
+
+      - name: Setup Pre-requisuites of TVK License Creation
```
[the remark this answers is not in the document]
Comment (msurudwad) [reply .github/workflows/plugin-packages.yml:157]: It'll be required to run log-collector integration tests where we install TVK & create backup, restore, etc.

**Pull request merged**